advisories/a82079.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Technical Advisory 82079
+advisory: A-82079
+summary: If a <code>CREATE MATERIALIZED VIEW</code> statement fails, all objects referenced in its <code>SELECT</code> query will be unusable.
+toc: true
+affected_versions: v21.2.0-v21.2.12, v22.1.0
+advisory_date: 2022-07-18
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+The problem can occur on any cluster running versions v21.2.0 to v21.2.12 or v22.1.0. If a `CREATE MATERIALIZED VIEW` statement fails, all the tables, views, sequences and types referenced in its `SELECT` query will become unusable.
+
+This problem can be diagnosed by running the following query:
+
+~~~ sql
+WITH
+	tables
+		AS (
+			SELECT
+				id, tab
+			FROM
+				(
+					SELECT
+						id,
+						crdb_internal.pb_to_json(
+							'cockroach.sql.sqlbase.Descriptor',
+							descriptor
+						)->'table'
+							AS tab
+					FROM
+						system.descriptor
+				)
+			WHERE
+				tab IS NOT NULL
+		),
+	old_depended_on_by_entries
+		AS (
+			SELECT
+				id, json_array_elements(tab->'dependedOnBy') AS entry
+			FROM
+				tables
+		),
+	ids_of_descriptors_that_need_upsert
+		AS (
+			SELECT
+				DISTINCT id
+			FROM
+				old_depended_on_by_entries
+			WHERE
+				(entry->>'id')::INT8 NOT IN (SELECT id FROM system.namespace)
+		)
+SELECT
+	count(*)
+FROM
+	ids_of_descriptors_that_need_upsert;
+~~~
+
+A zero result indicates the absence of the problem, and a non-zero result indicates the presence of the problem.
+
+## Statement
+
+This is resolved in CockroachDB by PR [#82087](https://github.com/cockroachdb/cockroach/pull/82087) which fixes the missing removal of back-references to the materialized view.
+
+The fix has been applied to maintenance releases of CockroachDB v21.2.13 and v22.1.1. This public issue is tracked by [#82079](https://github.com/cockroachdb/cockroach/pull/82079).
+
+## Mitigation
+
+Users of CockroachDB versions v21.2.0 to v21.2.12 or v22.1.0 are encouraged to upgrade to [v21.2.13](../releases/v21.2.html#v21-2-13) and [v22.1.1](../releases/v22.1.html#v22-1-1) or a later version.
```

Comment:
Add a sentence to the effect of: `Upgrading to a later maintenance release only prevents the problem from appearing, it will not fix an existing problem.`